packages/alfa-rules/test/sia-r12/rule.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import { test } from "@siteimprove/alfa-test";
+
+import { Document } from "@siteimprove/alfa-dom";
+
+import R12, { Outcomes } from "../../src/sia-r12/rule";
+
+import { evaluate } from "../common/evaluate";
+import { passed, failed, inapplicable } from "../common/outcome";
+
+test(`evaluates() passes a <button> with accessible name `, async (t) => {
+  const target = <button>My button</button>;
+
+  const document = Document.of([target]);
+
+  t.deepEqual(await evaluate(R12, { document }), [
+    passed(R12, target, {
+      1: Outcomes.HasName,
+    }),
+  ]);
+});
+
+test(`evaluates() passes an input element with accessible value attribute `, async (t) => {
```

Comment:
```suggestion
test(`evaluates() passes a submit button element with an accessible name given by the value attribute `, async (t) => {
```
* the rule is checking name of buttons, this `<input>` has a type of `submit` which makes it a "submit button" 🤷 We want the description to point out it is a button since that explains why this rule is testing it.
* "accessible name" is the technically correct term, so we also want to use it in descriptions.